setup.py

```diff
@@ -12,40 +12,85 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import os
 import pathlib
+import platform
+import subprocess
 
-from setuptools import setup
+import setuptools
+import setuptools.command.build_py
+from wheel.bdist_wheel import bdist_wheel as _bdist_wheel
+from wheel.bdist_wheel import get_platform
 
 here = pathlib.Path(__file__).parent.resolve()
-long_description = (here / "README.md").read_text(encoding="utf-8")
 
-setup(
+
+class bdist_wheel(_bdist_wheel):
+    def finalize_options(self):
+        self.plat_name = get_platform(None)  # force a platform tag
+        _bdist_wheel.finalize_options(self)
+
+
+class BuildPyCommand(setuptools.command.build_py.build_py):
+    """ Download a prebuilt version of livekit_ffi """
+
+    def run(self):
+
+        download_script = here / 'client-sdk-rust' / 'download_ffi.py'
+        cmd = ['python3', download_script.absolute(), '--output',
+               'livekit/resources']
+
+        # cibuildwheel is crosscompiling to arm64 on macos, make sure we download the
+        # right binary (kind of a hack here...)
+        if os.environ.get("CIBUILDWHEEL") == "1" \
+                and "arm64" in os.environ.get("ARCHFLAGS", ""):
+            cmd += ['--arch', 'arm64']
+
+        subprocess.run(cmd, capture_output=True, check=True)
+        setuptools.command.build_py.build_py.run(self)
+
+
+if platform.system() == "Linux":
+    libname = "liblivekit_ffi.so"
+elif platform.system() == "Darwin":
+    libname = "liblivekit_ffi.dylib"
+elif platform.system() == "Windows":
+    libname = "livekit_ffi.dll"
+
+
+setuptools.setup(
     name="livekit",
     version="0.2.3",
     description="LiveKit Python Client SDK for LiveKit",
-    long_description=long_description,
+    long_description=(here / "README.md").read_text(encoding="utf-8"),
     long_description_content_type="text/markdown",
     url="https://github.com/livekit/client-sdk-python",
+    cmdclass={
+        'bdist_wheel': bdist_wheel,
+        'build_py': BuildPyCommand,
+    },
     classifiers=[
-        "Development Status :: 3 - Alpha",
         "Intended Audience :: Developers",
         "License :: OSI Approved :: Apache Software License",
+        "Topic :: Multimedia :: Sound/Audio",
+        "Topic :: Multimedia :: Video",
+        "Topic :: Scientific/Engineering :: Artificial Intelligence",
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.7",
         "Programming Language :: Python :: 3.8",
         "Programming Language :: Python :: 3.9",
         "Programming Language :: Python :: 3.10",
         "Programming Language :: Python :: 3 :: Only",
     ],
-
-    keywords="webrtc, livekit",
+    keywords=["webrtc", "realtime", "audio", "video", "livekit"],
     license="Apache-2.0",
     packages=["livekit"],
-    python_requires=">=3.7, <4",
+    python_requires=">=3.7.0",
```

Comment:
do we really need 3.7? or is 3.6 ok too?

Reply:
We use dataclass, I just saw this is from Python 3.7: https://peps.python.org/pep-0557/